llvm-to-smt/generate_encodings.py

```diff
@@ -358,13 +358,8 @@ def mark_reg_unknown_memset_remove(verifier_c_filepath):
     input_file_line_iter = iter(input_file_lines)
     for i, line in enumerate(input_file_line_iter):
         if not memset_removal_done:
-            if r"static void __mark_reg_unknown(const struct bpf_verifier_env *env," in line:
-                next_line = next(input_file_line_iter)
-                line += next_line
-                next_next_line = next(input_file_line_iter)
-                line += next_next_line
-                if "struct bpf_reg_state *reg)" in next_line and "{" in next_next_line:
-                    func_start_found = True
+            if r"static void __mark_reg_unknown_imprecise(struct bpf_reg_state" in line:
```

Comment:
We could do a kernel version/commit based find-and-delete here. Kernels before [6efbde200bf3](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=6efbde200bf3cf2dbf6e7181893fed13a79c789b) would try to find memset starting from "__mark_reg_unknown", kernels after would try to find memset starting from "__mark_reg_unknown_imprecise"

Reply:
I can work on this! Not sure if I am able to push to this PR.

Reply:
Couldn't we look for `__mark_reg_unknown_imprecise` first and try to patch `__mark_reg_unknown` as a fallback? That would avoid the need to check the version/commit.

> Not sure if I am able to push to this PR.

You should be able to checkout my branch from my fork and push to it.
```
git remote add pchaigno git@github.com:pchaigno/agni
git checkout update-llvm-dep-14.0.0
...
```
You don't have full access to my fork but I did allow you for this specific PR :-)

Reply:
Got it, thanks! 

> Couldn't we look for __mark_reg_unknown_imprecise first and try to patch __mark_reg_unknown as a fallback? That would avoid the need to check the version/commit.

This sounds good to me! I was unsure if you are working on this. Let me know if you would like me to work on it.

Reply:
Go ahead! I'm going to Dagstuhl next week and still have to finish my presentation so I probably won't have much time until I'm back.